pgjdbc/src/main/java/org/postgresql/core/v3/SimpleParameterList.java

```diff
@@ -43,12 +48,53 @@
   private static final byte TEXT = 0;
   private static final byte BINARY = 4;
 
-  SimpleParameterList(int paramCount, @Nullable TypeTransferModeRegistry transferModeRegistry) {
+  private static final ParameterContext NO_PARAMETER_CONTEXT = ParameterContext.EMPTY_CONTEXT;
+
+  /**
+   * Marker object representing NULL; this distinguishes "parameter never set" from "parameter set
+   * to null".
+   */
+  private static final Object NULL_OBJECT = new Object();
+  private final @Nullable Map<String, Integer> paramNameIndex;
+  private final @Nullable List<String> paramNames;
+  private final @Nullable Object[] paramValues;
+  private final int[] paramTypes;
+  private final byte[] flags;
+  private final byte[] @Nullable [] encoded;
+  private final @Nullable TypeTransferModeRegistry transferModeRegistry;
+  private int pos = 0;
+
+  SimpleParameterList(int paramCount,
+      @Nullable TypeTransferModeRegistry transferModeRegistry) {
+    this(paramCount, transferModeRegistry, NO_PARAMETER_CONTEXT);
+  }
+
+  SimpleParameterList(int paramCount,
+      @Nullable TypeTransferModeRegistry transferModeRegistry,
+      ParameterContext parameterCtx
+  ) {
     this.paramValues = new Object[paramCount];
     this.paramTypes = new int[paramCount];
     this.encoded = new byte[paramCount][];
     this.flags = new byte[paramCount];
     this.transferModeRegistry = transferModeRegistry;
+
+    if (parameterCtx.hasNamedParameters()) {
+      // paramNameIndex will be used to perform index lookup in the various setter methods in
+      // PgPreparedStatement
+
+      this.paramNames = new ArrayList<>(paramCount);
```

Comment:
isn't this a copy of what is in parameterCtx placeholder names? Can we not just use what is in parameter context?